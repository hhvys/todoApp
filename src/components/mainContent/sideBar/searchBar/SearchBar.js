import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {toggleSideBar, searchQuery} from '../../../../actions/actionCreaters';
import SearchSymbol from '../../../logos/SearchSymbol';
import '../../../logos/logos.css';
import './searchBar.css';

const onQueryChange = (e, onInputChange) => {
	const query = e.target.value.trim();
	onInputChange(query);
};

let SearchBar = ({className, collapseSideBar, onInputChange}) => (
	<div className={className}>
		<a href="#" onClick={collapseSideBar}>
			<div></div>
			<div></div>
			<div></div>
		</a>
		<div className="search-input d-flex align-items-center pl-2">
			<input type="text" placeholder="Search" onChange={(e) => onQueryChange(e, onInputChange)}/>
		</div>
		<div className='search-symbol d-flex align-items-center justify-content-center'>
			<SearchSymbol />
		</div>
	</div>
)

SearchBar.propTypes = {
	className: PropTypes.string,
	collapseSideBar: PropTypes.func.isRequired
};

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
	collapseSideBar: () => dispatch(toggleSideBar()),
	onInputChange: (query) => {
		dispatch(searchQuery(query))
	}
});

SearchBar = connect(
	mapStateToProps,
	mapDispatchToProps
)(SearchBar);

export default SearchBar;