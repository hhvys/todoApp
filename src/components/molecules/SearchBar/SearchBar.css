.search__div {
    color: white;
}