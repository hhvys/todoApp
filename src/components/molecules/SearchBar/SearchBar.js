import React from 'react';
import styles from './SearchBar.mod.scss';
import Icon from '../../atoms/Icons/Icon';
import Input from '../../atoms/Input/Input';
import {LIST_TOGGLE, SEARCH_SYMBOL} from "../../atoms/Icons/constants";

const SearchBar = ({
			className,
			onInputChange,
			collapseSideBar,
			inputValue,
			...props
									 }) => (
			<div
				className={`${styles.search} full-width d-flex justify-content-center align-items-center ${className ? className : ''}`}
				{...props}>
				<div className={`${styles.collapse} d-flex align-items-center justify-content-center pl-2`}
						 onClick={collapseSideBar}>
					<Icon symbolType={LIST_TOGGLE}/>
				</div>
				<Input className={"pl-3"}
							 placeholder={'Search'}
							 onInputChange={onInputChange}
							 inputValue={inputValue}
				/>

		<div className={"d-flex align-items-center justify-content-center pr-2"}>
			<Icon
				className={styles.white}
				symbolType={SEARCH_SYMBOL}
			/>
		</div>
	</div>
);

export default SearchBar;
