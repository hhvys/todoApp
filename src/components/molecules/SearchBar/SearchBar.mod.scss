.search {
    color: white;
}

.collapse {
    cursor: pointer;
}

.white {
    fill: white;
}
