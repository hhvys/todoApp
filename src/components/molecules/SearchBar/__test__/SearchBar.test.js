import React from 'react';
import SearchBar from '../SearchBar';

describe('SearchBar', () => {

	let props;
	let mountedSearchBar;

	const searchBar = () => {
		if (!mountedSearchBar) {
			mountedSearchBar = shallow(
				<SearchBar {...props}/>
			);
		}
		return mountedSearchBar;
	};

	beforeEach(() => {
		props = {
			className: "className",
			onInputChange: undefined,
			collapseSideBar: undefined,
			inputValue: undefined,
		};
		mountedSearchBar = undefined;
	});

	it('should render', () => {
		const wrapperDiv = searchBar().find('div').first();
		expect(toJSON(wrapperDiv)).toMatchSnapshot();
	});

	it('should pass collapseSideBar prop to startLogo', () => {
		props.collapseSideBar = jest.fn();
		const wrapperDiv = searchBar().find('div').first();
		const startLogo = wrapperDiv.childAt(0);
		expect(toJSON(startLogo)).toMatchSnapshot();
	});

	it('should pass onInputChange and inputValue to Input', () => {
		props.onInputChange = jest.fn();
		props.inputValue = jest.fn();
		const wrapperDiv = searchBar().find('div').first();
		const startLogo = wrapperDiv.childAt(1);
		expect(toJSON(startLogo)).toMatchSnapshot();
	});

});
