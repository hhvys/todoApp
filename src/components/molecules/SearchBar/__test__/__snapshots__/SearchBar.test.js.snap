// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should pass collapseSideBar prop to startLogo 1`] = `
<div
  className="collapse d-flex align-items-center justify-content-center pl-2"
  onClick={[Function]}
>
  <Icon
    height={20}
    symbolType="LIST_TOGGLE"
    width={20}
  />
</div>
`;

exports[`SearchBar should pass onInputChange and inputValue to Input 1`] = `
<Input
  className="pl-3"
  inputValue={[Function]}
  onInputChange={[Function]}
  placeholder="Search"
/>
`;

exports[`SearchBar should render 1`] = `
<div
  className="search full-width d-flex justify-content-center align-items-center className"
>
  <div
    className="collapse d-flex align-items-center justify-content-center pl-2"
  >
    <Icon
      height={20}
      symbolType="LIST_TOGGLE"
      width={20}
    />
  </div>
  <Input
    className="pl-3"
    placeholder="Search"
  />
  <div
    className="d-flex align-items-center justify-content-center pr-2"
  >
    <Icon
      className="white"
      height={20}
      symbolType="SEARCH_SYMBOL"
      width={20}
    />
  </div>
</div>
`;
